packages/nx/src/command-line/add/add.ts

```diff
@@ -0,0 +1,139 @@
+import { exec } from 'child_process';
+import * as ora from 'ora';
+import { isAngularPluginInstalled } from '../../adapter/angular-json';
+import type { GeneratorsJsonEntry } from '../../config/misc-interfaces';
+import { logger } from '../../utils/logger';
+import { output } from '../../utils/output';
+import {
+  getPackageManagerCommand,
+  type PackageManagerCommands,
+} from '../../utils/package-manager';
+import { handleErrors } from '../../utils/params';
+import { getPluginCapabilities } from '../../utils/plugins';
+import { workspaceRoot } from '../../utils/workspace-root';
+import type { AddOptions } from './command-object';
+
+export function addHandler(args: AddOptions): Promise<void> {
+  if (args.verbose) {
+    process.env.NX_VERBOSE_LOGGING = 'true';
+  }
+  const isVerbose = process.env.NX_VERBOSE_LOGGING === 'true';
+
+  return handleErrors(isVerbose, async () => {
+    output.addNewline();
+
+    const pmc = getPackageManagerCommand();
+    const [pkgName, version] = parsePackageSpecifier(args.packageSpecifier);
+
+    await installPackage(pkgName, version, pmc);
+    await initializePlugin(pkgName, pmc);
+
+    output.success({
+      title: `Package ${pkgName} added successfully.`,
+    });
+  });
+}
+
+async function installPackage(
+  pkgName: string,
+  version: string,
+  packageManagerCommands: PackageManagerCommands
+): Promise<void> {
+  const spinner = ora(`Installing ${pkgName}@${version}...`);
+  spinner.start();
+
+  await new Promise<void>((resolve) =>
+    exec(
+      `${packageManagerCommands.addDev} ${pkgName}@${version}`,
+      (error, stdout) => {
+        if (error) {
+          spinner.fail();
+          output.addNewline();
+          logger.log(stdout);
+          process.exit(1);
+        }
+
+        return resolve();
+      }
+    )
+  );
+
+  spinner.succeed();
+}
```

Comment:
Can we update this to handle `.nx` installations? We'd just check if package.json doesn't exist and add the package to nxJson.installation.plugins, the actual installation would happen when we call the generate command later on or could be done ahead of time by running something like `nx --version` silently.

Reply:
Will do! Good catch. I forgot about it.